src/Stratis.Bitcoin.IntegrationTests.Common/Runners/StratisPosApiRunner.cs

```diff
@@ -31,7 +32,7 @@ public override void BuildNode()
                             .UseMempool()
                             .AddPowPosMining()
                             .UseWallet()
-                            .UseApi()
+                            .UseApi(option => option.CertificateStoreName = StoreName.My)
```

Comment:
Why is the property being set as an Action ? 

Would this read better :
```


public override void BuildNode()
{
    var settings = new NodeSettings(args: new string[] { "-conf=bitcoin.conf", "-datadir=" + this.DataFolder });
    var options = new ApiFeatureOptions(StoreName.My);

    this.FullNode = (FullNode)new FullNodeBuilder()
        .UseNodeSettings(settings)
        .UseBlockStore()
        .UsePowConsensus()
        .UseMempool()
        .AddMining()
        .UseWallet()
        .AddRPC()
        .UseApi(options)
        .MockIBD()
        .Build();
}

```

Then in ApiFeature : 

```
public static class ApiFeatureExtension
{
    public static IFullNodeBuilder UseApi(this IFullNodeBuilder fullNodeBuilder, ApiFeatureOptions options = null)
    {
        CertificateStore certificateStore = null;

        if (options != null)
            certificateStore = new CertificateStore(options.CertificateStoreName, StoreLocation.CurrentUser);

        fullNodeBuilder.ConfigureFeature(features =>
        {
            features
                .AddFeature<ApiFeature>()
                .FeatureServices(services =>
                {
                    services.AddSingleton(fullNodeBuilder);
                    services.AddSingleton(options);
                    services.AddSingleton<ApiSettings>();

                    if (certificateStore != null)
                        services.AddSingleton<ICertificateStore>(certificateStore);
                });
        });

        return fullNodeBuilder;
    }
}
```

Reply:
I just didn't feel like changing the option or apifeature class too much
If everybody feels like it is bad I will change no pb, let me know

Reply:
I was thinking it looks more consistent with how `NodeSettings` is setup.

Also if new functionality is introduced in `ApiFeatureOptions`, then the code would need to change to this anyway - maybe cross that bridge when it comes to it.